BuildingBlocks/src/BuildingBlocks.API/Mvc/ExceptionFilters/CustomExceptionFilter.cs

```diff
@@ -218,3 +216,72 @@ private IEnumerable<string> GetFormattedStackTrace(Exception exception)
             Regex.Matches(exception.StackTrace, "at .+").Select(m => m.Value.Trim());
     }
 }
+
+file static class LoggerExtensions
+{
+    private static readonly Action<ILogger, InfrastructureException, string, string, Exception> INFRASTRUCTURE_EXCEPTION =
+        LoggerMessage.Define<InfrastructureException, string, string>(
+            LogLevel.Information,
+            new EventId(560507, "ExceptionFilter.InfrastructureException"),
+            "An '{exception}' occurred. Error Code: '{code}'. Error message: '{message}'."
+        );
+
+    private static readonly Action<ILogger, ApplicationException, string, string, Exception> APPLICATION_EXCEPTION =
+        LoggerMessage.Define<ApplicationException, string, string>(
+            LogLevel.Information,
+            new EventId(437832, "ExceptionFilter.ApplicationException"),
+            "An '{exception}' occurred. Error Code: '{code}'. Error message: '{message}'."
+        );
+
+    private static readonly Action<ILogger, DomainException, string, string, Exception> DOMAIN_EXCEPTION =
+        LoggerMessage.Define<DomainException, string, string>(
+            LogLevel.Information,
+            new EventId(505278, "ExceptionFilter.DomainException"),
+            "An '{exception}' occurred. Error Code: '{code}'. Error message: '{message}'."
+        );
+
+    private static readonly Action<ILogger, string, Exception> REQUEST_BODY_TOO_LARGE =
+        LoggerMessage.Define<string>(
+            LogLevel.Information,
+            new EventId(938218, "ExceptionFilter.RequestBodyTooLarge"),
+            "'{error_code}': The body of the request is too large."
+        );
+
+    private static readonly Action<ILogger, Uri, Exception> ERROR_WHILE_PROCESSING_REQUEST_TO_URI =
+        LoggerMessage.Define<Uri>(
+            LogLevel.Error,
+            new EventId(259125, "ExceptionFilter.ErrorWhileProcessingRequestToUri"),
+            "Unexpected Error while processing request to '{uri}'."
+        );
+
+
+    public static void InfrastructureException(
+        this ILogger logger, InfrastructureException infrastructureException, string errorCode, string errorMessage)
+    {
+        INFRASTRUCTURE_EXCEPTION(logger, infrastructureException, errorCode, errorMessage, default!);
```

Comment:
You missed something. Before your change it was
```c#
_logger.LogInformation(
                    "An '{exception}' occurred. Error Code: '{code}'. Error message: '{message}'", nameof(InfrastructureException), infrastructureException.Code, infrastructureException.Message);
```

So only `nameof(InfrastructureException)` was passed as first parameter. Now you pass the whole exception.